test/parallel/test-error-format-list.js

```diff
@@ -0,0 +1,28 @@
+// Flags: --expose-internals
+'use strict';
+
+require('../common');
+const { strictEqual } = require('node:assert');
+const { formatList } = require('internal/errors');
+
+strictEqual(formatList([]), '');
+
+strictEqual(formatList([], 'or'), '');
+
+strictEqual(formatList(['apple']), 'apple');
+
+strictEqual(formatList(['apple'], 'or'), 'apple');
+
+strictEqual(formatList(['apple', 'banana']), 'apple and banana');
+
+strictEqual(formatList(['apple', 'banana'], 'or'), 'apple or banana');
+
+strictEqual(
+  formatList(['apple', 'banana', 'orange']),
+  'apple, banana, and orange'
+);
+
+strictEqual(
+  formatList(['apple', 'banana', 'orange'], 'or'),
+  'apple, banana, or orange'
+);
```

Comment:
nit: we could use the built-in for tests though:

```suggestion
}

if (common.hasIntl) {
  const fAnd = new Intl.ListFormat('en', { style: 'long', type: 'conjunction' });
  const fOr = new Intl.ListFormat('en', { style: 'long', type: 'disjunction' });
  
  const input = ['apple', 'banana', 'orange'];
  for (let i = 0; i < input.length; i++) {
    const slicedInput = input.slice(0, i);
    stringEqual(formatList(slicedInput), fAnd.format(slicedInput));
    stringEqual(formatList(slicedInput, 'or'), fOr.format(slicedInput));
  }
}